### PR TITLE
Add volser option to files list data-set command

### LINKED ISSUE #61: Allow specification of volser/volume option to list command
_From @PeteSwauger on May 30, 2018 13:34_

zosmf allows the specification of a volser.  Allowing volser provides access to uncatalogued data sets.

GET /zosmf/restfiles/ds/?dslevel=<filter-criteria>[&start=dsname]
where:
/zosmf/restfiles specifies the z/OS® data set and file REST interface
/ds indicates a data set request
?dslevel=Start of change<dataset-name-pattern>End of change[&volser=Start of change<volser>&start=<dsname>End of change] represents the query parameters used to qualify the request, such as a data set name and, optionally, a volume serial (VOLSER).

_Copied from original issue: gizafoundation/brightside#117_

### PR DESCRIPTION
**Pull request by t1m0thyj**
Resolves #61 

The `stringLengthRange` was added per IBM's z/OSMF doc:
> **volser** - A parameter that identifies the volume serials to be searched for data sets with names that match the specified dslevel parameter. The volume serial is one to six characters. You cannot use wildcard characters for this parameter. If you omit this parameter, the cataloged data set name is returned by default. If this parameter is specified, the data sets on the volume that match the dslevel pattern are returned.

### CODE REVIEW

**Review by gejohnston — APPROVED** [review]
Looks good.

File: packages/cli/src/zosfiles/list/List.options.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ export const ListOptions: { [key: string]: ICommandOptionDefinition } = {
         name: "volume-serial",
         aliases: ["vs"],
         description: strings.VOLUME,
-        type: "string"
+        type: "string",
+        stringLengthRange: [1, 6]  // tslint:disable-line:no-magic-numbers
```
Comment (zFernand0) [inline packages/cli/src/zosfiles/list/List.options.ts:32]: Wondering if we should revisit linting rules at some point.
:yum:

Comment (t1m0thyj) [reply packages/cli/src/zosfiles/list/List.options.ts:32]: Maybe when we migrate to ESLint we can revisit them 🙂

**Comment by zFernand0** [comment]
Thanks for resolving merge conflicts (if any) 😋

**Pull request merged**